File: app/src/main/java/cat/xojan/random1/domain/repository/DownloadPodcastRepository.java
package cat.xojan.random1.domain.repository;

import java.util.Set;

import cat.xojan.random1.domain.entities.Podcast;

public interface DownloadPodcastRepository {

    boolean addDownloadingPodcast(Podcast podcast);

    boolean deleteDownloadingPodcast(Podcast podcast);

    void setPodcastAsDownloaded(String audioId, String filePath);

    Set<Podcast> getDownloadingPodcasts();

    Set<Podcast> getDownloadedPodcasts();

    void deleteDownloadedPodcast(Podcast podcast);
}

File: app/src/main/java/cat/xojan/random1/domain/repository/DownloadPodcastRepository.kt
package cat.xojan.random1.domain.repository

import android.support.v4.media.MediaDescriptionCompat

interface DownloadPodcastRepository {

    fun addDownloadingPodcast(podcast: MediaDescriptionCompat): Boolean

    fun deleteDownloadingPodcast(podcast: MediaDescriptionCompat): Boolean

    fun setPodcastAsDownloaded(mediaId: String, filePath: String)

    fun getDownloadingPodcasts(): MutableSet<MediaDescriptionCompat>

    fun getDownloadedPodcasts(): Set<MediaDescriptionCompat>

    fun deleteDownloadedPodcast(podcast: MediaDescriptionCompat)

    fun getDownloadedPodcastTitle(mediaId: String): String?
}